samples/certs/Makefile

```diff
@@ -121,3 +128,57 @@ root-key.pem:
 	@echo "O = $(CITADEL_ORG)" >> $@
 	@echo "CN = $(CITADEL_CN)" >> $@
 	@echo "L = $(L:/=)" >> $@
+
+#------------------------------------------------------------------------
+##<name>-certs-wl: generate certificates for a virtual machine connected to the namespace `$NAMESPACE` using serviceAccount `$SERVICE_ACCOUNT` as well as intermediate certificates for K8s cluster `$NAME`
+.PHONY: %-certs-wl
+
+%-certs-wl: %/cert-chain.pem root-cert.pem %/workload-cert-chain.pem
+	@echo "Citadel inputs and workload certs stored in $(dir $<)"
+	@cp root-cert.pem $(dir $<)
+
+%/workload-cert-chain.pem: %/workload-cert.pem %/ca-cert.pem root-cert.pem
+	@echo "generating $@"
+	@cat $^ > $@
+
+%/workload-cert.pem: %/workload.csr
```

Comment:
I think the best practice was to create a separate intermediary CA for VMs.

The sample makefile should create a root CA, an intermediary CA to be deployed
in the cluster ( and for MC - separate intermediary for each clustser ), and one or more 
intermediary CAs for VMs.

Using the same intermediary for clustser and VMs is not ideal.

Reply:
+1